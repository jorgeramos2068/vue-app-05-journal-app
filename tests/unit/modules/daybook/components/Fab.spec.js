import { shallowMount } from '@vue/test-utils';
import Fab from '@/modules/daybook/components/Fab';

describe('Fab component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Fab);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should show the default icon', () => {
    expect(wrapper.find('i.fa-plus').exists()).toBeTruthy();
  });

  it('should show the icon by argument', () => {
    const testIcon = 'fa-something';
    const wrapper = shallowMount(Fab, {
      props: { icon: testIcon },
    });
    expect(wrapper.find(`i.${testIcon}`).exists()).toBeTruthy();
  });

  it('should emit the on-click event', () => {
    const button = wrapper.find('button');
    button.trigger('click');
    expect(wrapper.emitted('on-click').length).toBe(1);
  });
});
