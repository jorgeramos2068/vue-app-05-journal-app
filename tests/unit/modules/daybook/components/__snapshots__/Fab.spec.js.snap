// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab component should match the snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
